### PR TITLE
FFU-1 Delete sub attribute EP

### PR DESCRIPTION
**Pull request by RamsesDijkstra**
Change interface attribute name to make it consistent with the rest, added delete EP (bij making subattributes nil)

### CODE REVIEW

File: universe/node/api_space_attributes.go [hunk 1]
```diff
@@ -197,3 +197,55 @@ func (n *Node) apiSetSpaceSubAttribute(c *gin.Context) {
 
 	c.JSON(http.StatusCreated, out)
 }
+
+func (n *Node) apiDeleteSpaceSubAttribute(c *gin.Context) {
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: please look into this EP and change your accordingly

File: universe/node/api_space_attributes.go [hunk 2]
```diff
@@ -194,3 +194,55 @@ func (n *Node) apiSetSpaceAttributeSubValue(c *gin.Context) {
...
+		err := errors.WithMessage(err, "Node: apiDeleteSpaceAttributeSubValue: failed to parse space id")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_space_id", err, n.log)
+		return
+	}
+
+	if err := c.ShouldBindQuery(&inQuery); err != nil {
+		err := errors.WithMessage(err, "Node: apiDeleteSpaceAttributeSubValue: failed to bind query")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_query", err, n.log)
+		return
+	}
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: please move it before parsing space id

**Review by cnaize — CHANGES_REQUESTED** [review]

File: universe/node/api_space_attributes.go [hunk 3]
```diff
@@ -194,3 +194,55 @@ func (n *Node) apiSetSpaceAttributeSubValue(c *gin.Context) {
...
+
+	attributeID := entry.NewAttributeID(pluginID, inQuery.AttributeName)
+
+	modifyFn := func(current *entry.AttributePayload) (*entry.AttributePayload, error) {
+		(*current.Value)[inQuery.SubAttributeKey] = nil
+
+		return current, nil
+	}
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: fix this please

**Review by cnaize — CHANGES_REQUESTED** [review]

File: universe/node/api_space_attributes.go (hunk 2 above)
Comment (RamsesDijkstra) [reply universe/node/api_space_attributes.go]: moved

File: universe/node/api_space_attributes.go [hunk 4]
```diff
@@ -194,3 +194,59 @@ func (n *Node) apiSetSpaceAttributeSubValue(c *gin.Context) {
 
 	c.JSON(http.StatusCreated, out)
 }
+
+func (n *Node) apiDeleteSpaceAttributeSubValue(c *gin.Context) {
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: please rename to `remove`, we use everywhere

File: universe/node/api_space_attributes.go [hunk 5]
```diff
@@ -194,3 +194,59 @@ func (n *Node) apiSetSpaceAttributeSubValue(c *gin.Context) {
...
+		if current == nil {
+			return current, nil
+		}
+
+		delete(*current, inQuery.SubAttributeKey)
+
+		return current, nil
+	}
+
+	err = space.UpdateSpaceAttributeValue(attributeID, modifyFn, true)
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: please check the error on the same line, you have enough space

File: universe/node/api_space_attributes.go [hunk 6]
```diff
@@ -194,3 +194,59 @@ func (n *Node) apiSetSpaceAttributeSubValue(c *gin.Context) {
...
+	if err != nil {
+		err = errors.WithMessage(err, "Node: apiDeleteSpaceAttributeSubValue: failed to upsert space attribute")
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: and fix please this message

File: universe/node/api_space_attributes.go [hunk 7]
```diff
@@ -194,3 +194,59 @@ func (n *Node) apiSetSpaceAttributeSubValue(c *gin.Context) {
...
+		PluginID        string `form:"plugin_id" binding:"required"`
+		AttributeName   string `form:"attribute_name" binding:"required"`
+		SubAttributeKey string `form:"sub_attribute_key" binding:"required"`
+	}{}
+
+	if err := c.ShouldBindQuery(&inQuery); err != nil {
+		err := errors.WithMessage(err, "Node: apiDeleteSpaceAttributeSubValue: failed to bind query")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_query", err, n.log)
+		return
+	}
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: please change it to parse body

**Review by cnaize — CHANGES_REQUESTED** [review]

File: universe/node/api_space_attributes.go [hunk 8]
```diff
@@ -194,3 +194,58 @@ func (n *Node) apiSetSpaceAttributeSubValue(c *gin.Context) {
...
+		}
+
+		delete(*current, inQuery.SubAttributeKey)
+
+		return current, nil
+	}
+
+	if err := space.UpdateSpaceAttributeValue(attributeID, modifyFn, true); err != nil {
+		err = errors.WithMessage(err, "Node: apiRemoveSpaceAttributeSubValue: failed to remove space attribute")
+		api.AbortRequest(c, http.StatusInternalServerError, "failed_to_upsert", err, n.log)
```
Comment (cnaize) [inline universe/node/api_space_attributes.go]: and here is invalid error reason

Comment (RamsesDijkstra) [reply universe/node/api_space_attributes.go]: ill make it failed_to_update since thats what it is calling "UpdateSpaceAttributeValue"

**Review by cnaize — APPROVED** [review]

**Pull request merged**